js_modules/dagit/packages/core/src/app/ExecutionSessionStorage.tsx

```diff
@@ -163,24 +170,27 @@ export function useExecutionSessionStorage(
 ): StorageHook {
   const {basePath} = React.useContext(AppContext);
 
-  const oldNamespace = `${repoAddress.name}.${pipelineOrJobName}`;
-  const oldData = getStorageDataForNamespace(oldNamespace);
-
-  const namespace = `${basePath}-${repoAddress.location}-${repoAddress.name}-${pipelineOrJobName}`;
-  const [version, setVersion] = React.useState<number>(0);
+  const namespace = makeNamespace(basePath, repoAddress, pipelineOrJobName);
+  const [_, setVersion] = React.useState<number>(0);
 
-  const onSave = (newData: IStorageData) => {
-    writeStorageDataForNamespace(namespace, newData);
-    setVersion(version + 1); // trigger a React render
-  };
+  const onSave = React.useCallback(
+    (newData: IStorageData) => {
+      writeStorageDataForNamespace(namespace, newData);
+      setVersion((current) => current + 1); // trigger a React render
+    },
+    [namespace],
+  );
 
-  // TODO: Remove this migration logic in a few patches when we know the old namespace is likely no longer being used
-  const oldDataMigrated = React.useRef(false);
-  if (oldData && !oldDataMigrated.current) {
```

Comment:
These changes all look good to me - the core issue was that `getStorageDataForNamespace(oldNamespace)`  always returns a truthy value (`{sessions: {}, current: ''}`), even if no saved localstorage data is present, so this was running every time the component mounted?

Reply:
yep